From 506ad5d35bd3d9ddf832cae823947e0de111dafa Mon Sep 17 00:00:00 2001
From: Christian Marangi <ansuelsmth@gmail.com>
Date: Thu, 20 Nov 2025 11:07:23 +0100
Subject: [PATCH 1/3] iax: client: use ssize_t for sendto_t typedef

Linux use ssize_t as the return value of a sendto function. use that to
fix compilation error on new GCC version.

iax/iax.c:418:27: error: assignment to 'sendto_t' {aka 'int (*)(int,  const void *, long unsigned int,  int,  const struct sockaddr *, unsigned int)'} from incompatible pointer type 'ssize_t (*)(int,  const void *, size_t,  int,  const struct sockaddr *, socklen_t)' {aka 'long int (*)(int,  const void *, long unsigned int,  int,  const struct sockaddr *, unsigned int)'} [-Wincompatible-pointer-types]

Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
---
 iax/iax-client.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/iax/iax-client.h
+++ b/iax/iax-client.h
@@ -77,7 +77,7 @@ struct iax_session;
 #ifdef WIN32
 typedef int PASCAL (*sendto_t)(SOCKET, const char *, int, int, const struct sockaddr *, int);
 #else
-typedef int (*sendto_t)(int, const void *, size_t, int, const struct sockaddr *, socklen_t);
+typedef ssize_t (*sendto_t)(int, const void *, size_t, int, const struct sockaddr *, socklen_t);
 #endif
 
 struct iax_event {
